src/index/blockfilterindex.h

```diff
@@ -49,9 +50,9 @@ class BlockFilterIndex final : public BaseIndex
 
     bool Rewind(const CBlockIndex* current_tip, const CBlockIndex* new_tip) override;
 
-    BaseIndex::DB& GetDB() const override { return *m_db; }
+    BaseIndex::DB& GetDB() const LIFETIMEBOUND override { return *m_db; }
 
-    const char* GetName() const override { return m_name.c_str(); }
+    const char* GetName() const LIFETIMEBOUND override { return m_name.c_str(); }
```

Comment:
unrelated, as it requires a ton of changes, but it would be nice to change this to return a `const std::string&`

Reply:
Agreed, I did a quick draft and it doesn't seem like a big lift actually. Will follow up with PR shortly.

Reply:
Follow up in https://github.com/bitcoin/bitcoin/pull/25971